django/http/multipartparser.py

```diff
@@ -49,6 +49,8 @@ class MultiPartParser:
     and returns a tuple of ``(MultiValueDict(POST), MultiValueDict(FILES))``.
     """
 
+    boundary_re = re.compile(b"[ -~]{0,200}[!-~]")
```

Comment:
maybe `_lazy_re_compile` ? also use `r`-sring so text editors highlight it as a regex?

Reply:
What's the benefit of `_lazy_re_compile` here? It's compiled once on import no? 🤔 
(This is a question, not an opinion. :)



(https://github.com/django/django/commit/fa85411701f7c5ef5802f377a1e5314144be48e5)

Reply:
> It's compiled once on import no?

Yes, but do we need to compile it at all? In most of cases it's not necessary because `boundary` is already parsed in `opts`.

Reply:
> It's compiled once on import no?

We made all module-level regexes to save import time, starting with 2bb1027d6bcdad59624a9d08701e0d2e4a9c0ba7. Often modules are imported but the relevant code paths aren't used.

Perhaps that's not so relevant with such short regexes - I benchmarked it as taking about 400ns to compile, which is not much.

But good point from Mariusz, if there's no need that's even better.

Reply:
OK, updated. Thanks both! 👍